kernelci/cli/config.py

```diff
@@ -75,3 +77,127 @@ def dump(section, config, indent, recursive):
     else:
         echo = click.echo_via_pager if recursive else click.echo
         echo(yaml.dump(data, indent=indent))
+
+
+def validate_rules(node, rules):
+    helper = kernelci.api.helper.APIHelper(None)
+    if helper.should_create_node(rules, node):
+        return True
+    else:
+        return False
+
+
+def compare_builds(merged_data):
+    """
+    Compare kbuilds and print builds with identical params
+    """
+    r = ""
+    jobs = merged_data.get("jobs")
+    if not jobs:
+        click.echo("No jobs found in the merged data, "
+                   "maybe you need to add parameter "
+                   "-c path/kernelci-pipeline/config?")
+        sys.exit(1)
+    kbuilds_list = []
+    for job in jobs:
+        if jobs[job].get("kind") == "kbuild":
+            kbuilds_list.append(job)
+
+    kbuilds_dict = {}
+    import json
+    for kbuild in kbuilds_list:
+        params = jobs[kbuild].get("params", {})
+        # Convert params to a hashable type by serializing to JSON
+        key = json.dumps(params, sort_keys=True)
+        if key not in kbuilds_dict:
+            kbuilds_dict[key] = []
+        kbuilds_dict[key].append(kbuild)
+
+    # print builds with identical params
+    for params, kbuild_list in kbuilds_dict.items():
+        if len(kbuild_list) > 1:
+            r += f"Params {params}: {kbuild_list},"
+
+    return r
+
+
+# pylint: disable=too-many-branches disable=too-many-locals
+def do_forecast(merged_data):
+    """
+    We will simulate checkout event on each tree/branch
+    and try to build list of builds/tests it will run
+    """
+    checkouts = []
+    build_configs = merged_data.get("build_configs", {})
+    for bcfg in build_configs:
+        data = build_configs[bcfg]
+        if not data.get("architectures"):
+            data["architectures"] = None
+        checkouts.append(data)
+
+    # sort checkouts by tree and branch
+    checkouts.sort(key=lambda x: (x.get("tree", ""), x.get("branch", "")))
+
+    # iterate over checkouts
+    for checkout in checkouts:
+        checkout["kbuilds"] = []
+        # iterate over events (jobs)
+        jobs = merged_data.get("scheduler", [])
```

Comment:
Does it include configs from different scheduler files such as `scheduler-cip.yaml` and `scheduler-chromeos.yaml`?